internal/commands/app/create.go

```diff
@@ -98,20 +101,39 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 		return err
 	}
 
-	var dsCluster dataSourceCluster
-	if cmd.inputs.Cluster != "" {
-		dsCluster, err = cmd.inputs.resolveCluster(clients.Atlas, groupID)
+	var nonExistingDataSources []string
+
+	var dsClusters []dataSourceCluster
```

Comment:
[meganit] micro-optimization here...

if you `var dsClustersMissing []string` alongside this here (and do the same with `var dsDatalakesMissing []string`), then further down (around L#126 right here) you could initialize `nonExistingDataSources` with `make([]string, 0, len(dsClustersMissing) + len(dsDatalakesMissing))` (and ultimately preventing any re-allocations with `nonExistingDataSources`)